src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -64,8 +64,10 @@ protected function generateUrl($route, $parameters = array(), $referenceType = U
      */
     protected function forward($controller, array $path = array(), array $query = array())
     {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $path['_forwarded'] = $request;
```

Comment:
@stof It appears to me as the simplest way to keep the forward trace, fortunately it's just an object reference, any suggestion ? maybe pass only the attributes bag ?